src/openms/source/ANALYSIS/QUANTITATION/IsotopeLabelingMDVs.cpp

```diff
@@ -0,0 +1,327 @@
+        {
+          intensities_vec.push_back(it->getMetaValue(feature_name));
+        }
+        std::vector<OpenMS::Peak2D::IntensityType>::iterator max_it = std::max_element(intensities_vec.begin(), intensities_vec.end());
+        double measured_feature_max = intensities_vec[std::distance(intensities_vec.begin(), max_it)];
+          
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (measured_feature_max != 0.0)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / measured_feature_max);
+          }
+        }
+      }
+    }
+    else if (mass_intensity_type == "norm_sum")
+    {
+      if (feature_name == "intensity")
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += it->getIntensity();
+        }
+        
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          normalized_feature.setMetaValue((it - measured_feature_subordinates.begin()), (it->getIntensity() / feature_peak_apex_intensity_sum));
+        }
+      }
+      // for every other case where feature_name isn't 'intensity', i.e. 'peak_apex_int'
+      else
+      {
+        OpenMS::Peak2D::IntensityType feature_peak_apex_intensity_sum = 0.0;
+        std::vector<OpenMS::Peak2D::IntensityType> intensities_vec;
+        for (auto it = measured_feature_subordinates.begin(); it != measured_feature_subordinates.end(); it++)
+        {
+          feature_peak_apex_intensity_sum += (Peak2D::IntensityType)it->getMetaValue(feature_name);
+        }
+
+        for (size_t i = 0; i < normalized_feature.getSubordinates().size(); ++i)
+        {
+          if (feature_peak_apex_intensity_sum != 0.0)
+          {
+            normalized_feature.getSubordinates().at(i).setIntensity((OpenMS::Peak2D::IntensityType)measured_feature_subordinates.at(i).getMetaValue(feature_name) / feature_peak_apex_intensity_sum);
+          }
+        }
+      }
+    }
+  }
+
+  void IsotopeLabelingMDVs::calculateMDVs(
+    const FeatureMap& measured_featureMap, FeatureMap& normalized_featureMap,
+    const String& mass_intensity_type, const String& feature_name)
+  {
+    for (const Feature& feature : measured_featureMap)
+    {
+      Feature normalized_feature;
+      calculateMDV(feature, normalized_feature, mass_intensity_type, feature_name);
+      normalized_featureMap.push_back(normalized_feature);
+    }
+  }
+
+  const std::unordered_map<std::string, std::vector<std::vector<double>> > IsotopeLabelingMDVs::correction_matrices_ =
```

Comment:
I think you can get away with
```
const std::unordered_map<std::string, std::vector<std::vector<double>> > correction_matrices_ {
      { "TBDMS", {{0.8213, 0.1053, 0.0734, 0.0000},
                  {0.8420, 0.0963, 0.0617, 0.0000},
                  {0.8466, 0.0957, 0.0343, 0.0233},
                  {0.8484, 0.0954, 0.0337, 0.0225}}
      }
};
```

and if you make this thing static, then all above functions can be static as well (if DefaultParamHandler can be removed, which I think is possible).

To ease usability, you could replace the  "TBDMS" `string` key with an ENUM because:
 - the allowed values are immediately clear
 - accidentally passing an invalid value is not possible

i.e.
```
enum class DerivAgent { 
  TBDMS 
};
```

Reply:
thanks for your suggestion, it has been updated to accommodate the ENUM instead of passing in a string for the derivatization agent.